test/functional/tool_wallet.py

```diff
@@ -545,6 +549,36 @@ def test_dump_unclean_lsns(self):
         self.stop_node(0)
         self.assert_tool_output("The dumpfile may contain private keys. To ensure the safety of your Bitcoin, do not share the dumpfile.\n", "-wallet=unclean_lsn", f"-dumpfile={wallet_dump}", "dump")
 
+    def test_compare_legacy_dump_with_framework_bdb_parser(self):
+        self.log.info("Verify that legacy wallet database dump matches the one from the test framework's BDB parser")
+        wallet_name = "bdb_ro_test"
+        self.start_node(0)
+        # add some really large labels (above twice the largest valid page size) to create BDB overflow pages
+        self.nodes[0].createwallet(wallet_name)
+        wallet_rpc = self.nodes[0].get_wallet_rpc(wallet_name)
+        for i in range(10):
+            large_label = ''.join([random.choice(string.ascii_letters) for _ in range(150000)])
+            wallet_rpc.setlabel(getnewdestination()[2], large_label)
+        self.stop_node(0)
+
+        wallet_dumpfile = self.nodes[0].datadir_path / "bdb_ro_test.dump"
+        args = ["-wallet={}".format(wallet_name), "-dumpfile={}".format(wallet_dumpfile), "dump"]
+        p = self.bitcoin_wallet_process(*args)
+        _, _ = p.communicate()
+        assert_equal(p.poll(), 0)
```

Comment:
In ad0a8ffceacfd5ce37826b2882b0e51a14c5a6bf "test: compare BDB dumps of test framework parser and wallet tool"

Could use `self.assert_tool_output`, see how the other calls to `dump` are done in this test.